packages/xo-server/src/xapi/mixins/storage.js

```diff
@@ -112,4 +112,26 @@ export default {
 
     return (await this.barrier(srRef)).uuid
   },
+
+  // This function helps to reattach a forgotten NFS/iSCSI SR
+  async reattachSr({ uuid, nameLabel, nameDescription, type, deviceConfig }) {
+    const srRef = await this.call('SR.introduce', uuid, nameLabel, nameDescription, type, 'user', true, {})
+
+    // XenCenter SR reattach:
+    // https://github.com/xenserver/xenadmin/blob/90e6cb0dc950ce747b7b6b689b0ad00cf28898d2/XenModel/Actions/SR/SrReattachAction.cs#L77-L99
+    const hosts = filter(this.objects.all, object => object.$type === 'host')
+    await Promise.all(
+      hosts.map(async host => {
+        const pbdRef = await this.call('PBD.create', {
+          host: host.$ref,
```

Comment:
If a `PBD` creation or plug fails, the `reattachSr` operation will fail without cleaning the created resources.